Automattic/jetpack
# Packages: Move sync to a classmapped package

This PR moves sync to a classmapped package as a preparatory task, so we can continue working on it and wrap it as an independent package.

#### Changes proposed in this Pull Request:
* Move sync to a classmapped package

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
Part of the Jetpack DNA project - p1HpG7-70O-p2

#### Testing instructions:
* Checkout this branch.
* Run `rm -rf ./vendor` inside Jetpack.
* Run `composer install` inside Jetpack.
* Verify that pass.
* Verify CI passes on this PR and 100% is green.
* Smoke test Jetpack to verify it still works well in both the admin and the frontend.
* Smoke test Jetpack sync (full sync, incremental sync)

#### Proposed changelog entry for your changes:
* Move sync to a classmapped package.
